aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java

```diff
@@ -0,0 +1,200 @@
+        asElasticsearchApiUser()
+            .when()
+            .delete(CASE_INDEX_NAME)
+            .then()
+            .statusCode(200)
+            .body("acknowledged", equalTo(true));
+    }
+
+    @Nested
+    @DisplayName("successful run")
+    class SuccessfulRun {
+
+        @Test
+        @DisplayName("should import a valid definition multiple times and create Elasticsearch index, alias and field mappings")
+        void shouldImportValidDefinitionMultipleTimes() {
+            // invoke definition import
+            asAutoTestImporter().get()
+                .given()
+                .multiPart(new File(DEFINITION_FILE))
+                .expect()
+                .statusCode(201)
+                .when()
+                .post("/import");
+
+            verifyIndexAndFieldMappings(false);
+
+            // invoke definition import second time
+            asAutoTestImporter().get()
+                .given()
+                .multiPart(new File(DEFINITION_FILE_WITH_NEW_FIELD))
+                .expect()
+                .statusCode(201)
+                .when()
+                .post("/import");
+
+            verifyIndexAndFieldMappings(true);
+        }
+
+    }
+
+    @Nested
+    @DisplayName("Failed run")
+    class FailedRun {
+
+        @Test
+        @DisplayName("should not import definition on Elasticsearch error")
+        void shouldNotImportDefinitionOnEsError() {
+            // invoke definition import
+            asAutoTestImporter().get()
+                .given()
+                .multiPart(new File(DEFINITION_FILE))
+                .expect()
+                .statusCode(201)
+                .when()
+                .post("/import");
+
+            verifyIndexAndFieldMappings(false);
+
+            // remove alias mapping to simulate failed scenario
+            removeAlias();
+
+            // invoking definition import second time should throw exception as alias does not exist
+            asAutoTestImporter().get()
+                .given()
+                .multiPart(new File(DEFINITION_FILE_WITH_NEW_FIELD))
+                .expect()
+                .statusCode(400)
+                .when()
+                .post("/import");
+        }
+
+        private void removeAlias() {
+            asElasticsearchApiUser()
+                .given()
+                .contentType(ContentType.JSON)
+                .body("{\"actions\":[{\"remove\":{\"index\":\"" + CASE_INDEX_NAME + "\",\"alias\":\"" + CASE_INDEX_ALIAS + "\"}}]}")
+                .when()
+                .post("_aliases")
+                .then()
+                .statusCode(200);
+        }
+
+    }
+
+    private void verifyIndexAndFieldMappings(boolean verifyNewFields) {
+        ValidatableResponse response = asElasticsearchApiUser()
+            .when()
+            .get(CASE_INDEX_ALIAS)
+            .then()
+            .statusCode(200);
+
+        verifyIndexAndAlias(response);
+        verifyMetadataFields(response);
```

Comment:
we already have integration tests that test the mapping. I would say we can then remove `verifyMetadataFields` and `verifyCaseDataFields`